MonoGame.Framework/Audio/DynamicSoundEffectInstance.XAudio.cs

```diff
@@ -91,9 +103,11 @@ private void OnBufferEnd(IntPtr obj)
             {
                 var buffer = _queuedBuffers.Dequeue();
                 buffer.Stream.Dispose();
+                _bufferPool.Return(_pooledBuffers.Dequeue());
```

Comment:
I think we should lock the queues to get rid of the race condition here. At least here (for XAudio) since this function is run from another thread.